src/components/shared/layout/Navbar.tsx

```diff
@@ -0,0 +1,138 @@
+};
+
+const Navbar = (props) => {
+  const { simple, grey, currentLocation, projects } = props;
+
+  const overlay = useOverlay();
+
+  const renderMobileLink = (
+    url,
+    label,
+    klass = "menu-link mobile-menu-link"
+  ) => {
+    if (url === currentLocation)
+      return (
+        <li className={klass} key={`mobile-menu-link-${label}`}>
+          <a onClick={overlay.hide}> {label} </a>
+        </li>
+      );
+
+    return (
+      <li className={klass} key={`mobile-menu-link-${label}`}>
+        <Link to={url}> {label} </Link>
+      </li>
+    );
+  };
+
+  const navbarBackButtonText = `Back to ${projects ? "projects" : "homepage"}`;
+
+  return (
+    <div className={`menu-container ${grey ? "grey" : ""}`}>
+      {simple ? (
+        <div className="menu">
+          <Link to="/">
+            <LogoSvg alt="Brains and Beards logo" />
+          </Link>
+          <div className="back-to-homepage">
+            <ChevronBackSvg className="reverse" alt="Back arrow" />
+            {projects
+              ? renderLink("/projects", navbarBackButtonText)
+              : renderLink("/", navbarBackButtonText)}
+          </div>
+        </div>
+      ) : (
+        <div className="menu">
+          <Link to="/">
+            <LogoSvg alt="Brains and Beards logo" />
+          </Link>
+
+          <ul className="menu-links desktop-only">
+            {[
+              { url: "/", label: "Home" },
+              { url: "/projects", label: "Projects" },
+              { url: "/about-us", label: "About us" },
+              { url: "/team", label: "Team" },
+              { url: "/jobs", label: "Careers" },
+              { url: "/blog", label: "Blog" },
+            ].map((link) => renderLink(link.url, link.label))}
+            <li className="menu-link">
+              <a href="https://podcast.brainsandbeards.com"> Podcast </a>
+            </li>
+          </ul>
+          <ul className="menu-estimate-link desktop-only">
+            {renderLink(
+              "/estimate-project",
+              "Estimate project",
+              "menu-link button menu-estimate-button"
+            )}
+          </ul>
+          <div onClick={overlay.show}>
+            <HamburgerSvg className="mobile-only" alt="menu" />
+          </div>
+        </div>
+      )}
+
+      {overlay.isShown && (
+        <div className="mobile-only menu-overlay">
+          <div className="mobile-menu-content">
+            <div className="menu">
+              <LogoSvg alt="Brains and Beards logo" />
+              <div onClick={overlay.hide}>
+                <CloseSvg className="mobile-only" alt="close" />
+              </div>
+            </div>
+            <ul>
+              {[
+                { url: "/", label: "Home" },
+                { url: "/projects", label: "Projects" },
+                { url: "/about-us", label: "About us" },
+                { url: "/team", label: "Team" },
+                { url: "/jobs", label: "Careers" },
```

Comment:
Something wrong with this redirect

> 404 Not found
> You just hit a route that doesn't exist... the sadness.

Reply:
I went through all other links, including project case studies and they all look good apart from this "Careers" one. 👍

I have not checked blog posts yet

Reply:
I've extracted the navbar link rendering to a separate component. Now it works as expected, the code is a bit cleaner, and I learnt what's going on there. (Apparently Gatsby has special JS handling for Link elements and treats them as internal.)